enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/RegisteredRewriter.scala

```diff
@@ -157,12 +157,16 @@ class RegisteredRewriter(tokenContext: TokenContext) {
         }
 
       case Variable(k) =>
-        pipelineInformation(k) match {
-          case LongSlot(offset, false, CTNode, name) => NodeFromRegister(offset, name)
-          case LongSlot(offset, true, CTNode, name) => NullCheck(offset, NodeFromRegister(offset, name))
-          case LongSlot(offset, false, CTRelationship, name) => RelationshipFromRegister(offset, name)
-          case LongSlot(offset, true, CTRelationship, name) => NullCheck(offset, RelationshipFromRegister(offset, name))
-          case RefSlot(offset, _, _, _) => ReferenceFromRegister(offset)
+        pipelineInformation.get(k) match {
```

Comment:
Instead of `match`ing on an `Option`, you could use map and getOrElse:

```
val x = pipelineInformation.get(k).map {
     case LongSlot(offset, false, CTNode, name) => NodeFromRegister(offset, name)
            case LongSlot(offset, true, CTNode, name) => NullCheck(offset, NodeFromRegister(offset, name))
            case LongSlot(offset, false, CTRelationship, name) => RelationshipFromRegister(offset, name)
            case LongSlot(offset, true, CTRelationship, name) => NullCheck(offset, RelationshipFromRegister(offset, name))
            case RefSlot(offset, _, _, _) => ReferenceFromRegister(offset)
            case _ =>
              throw new CantCompileQueryException("Unknown type for `" + k + "` in the pipeline information")
    }
x.getOrElse(throw new CantCompileQueryException("Did not find `" + k + "` in the pipeline information"))
```